### PR TITLE
Hubs/Scopes Merge 41b - Use COMBINED as default scope type for callbacks

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

This is an optional PR which we may or may not merge.

## :bulb: Motivation and Context

Using `CombinedScopeView`  as default scope for callbacks used in e.g. `configureScope()` or `withScope()` should be closer to what we did when there was only a single scope.

However, this also means there can be writes that manipulate e.g. global scope. One example where this is the case would be manipulating an object on `Contexts` where `CombinedContextsView` may hand back an object instance for let's say `Device` that lives on global scope. If you now change this `Device` object, global scope is changed.

For `configureScope` changing global scope may be OK but for `withScope` customers would likely expect it to not affect anything outside the callback. Having `configureScope` and `withScope` behave differently would also be odd.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Scopes.java [hunk 1]
```diff
@@ -677,7 +679,7 @@ public void configureScope(
               "Instance is disabled and this 'configureScope' call is a no-op.");
     } else {
       try {
-        callback.run(combinedScope.getSpecificScope(scopeType));
+        callback.run(combinedScope.getSpecificScope(scopeType, true));
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/Scopes.java:682]: Is there a specific reason for the new boolean param in `getSpecificScope`? Couldn't we just use `combinedScope` here? Or do we expect there to be more callbacks that would use this in the future?

File: sentry/src/main/java/io/sentry/Scopes.java [hunk 2]
```diff
@@ -636,7 +636,8 @@ public void withScope(final @NotNull ScopeCallback callback) {
     } else {
       final @NotNull IScopes forkedScopes = forkedCurrentScope("withScope");
       try (final @NotNull ISentryLifecycleToken ignored = forkedScopes.makeCurrent()) {
-        callback.run(forkedScopes.getScope());
+        //        callback.run(forkedScopes.getScope());
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/Scopes.java:639]: Commented line can be removed

File: sentry/src/main/java/io/sentry/Scopes.java [hunk 3]
```diff
@@ -657,7 +658,8 @@ public void withIsolationScope(final @NotNull ScopeCallback callback) {
     } else {
       final @NotNull IScopes forkedScopes = forkedScopes("withIsolationScope");
       try (final @NotNull ISentryLifecycleToken ignored = forkedScopes.makeCurrent()) {
-        callback.run(forkedScopes.getIsolationScope());
+        //        callback.run(forkedScopes.getIsolationScope());
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/Scopes.java:661]: Commented line can be removed

File: sentry/src/main/java/io/sentry/Scopes.java (hunk 1 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/Scopes.java:682]: If we pass in `ScopeType.COMBINED` here, we'd always use combined scope. If a user passes a specific scope into `configureScope` that scope should be used. We only want to change the default scope type for the callbacks not always use combined scope.

**Comment by adinauer** [comment]
Due to reasons listed in PR description, we're not going ahead with this. Closing this PR.

**Pull request closed without merging**